poker/src/components/ConnectToLobby/ConnectToLobby.tsx

```diff
@@ -25,79 +26,85 @@ interface Props {
-    console.log('room:', room);
-    console.log('url', url);
-
-    let userRole = ROLES.USER;
-    if (observer) userRole = ROLES.OBSERVER;
-    else if (isAdmin) userRole = ROLES.ADMIN;
-
-    const user: Member = {
-      userId: socket.id,
-      firstName,
-      lastName,
-      job: jobPosition,
-      role: userRole,
-    };
-
-    const callback = (response: ResponseFromSocket): void => {
-      console.log(response);
-
-      const { eventName, code, error: responseError, data } = response;
-
-      // eslint-disable-next-line no-console
-      if (responseError) console.log(`${eventName}: ${code}: ${responseError}`);
-      else {
-        const { user: responseUser } = data;
-        dispatch(setCurrentRoom(responseUser.room));
-        const userToRedux = {
-          userId: responseUser.userId,
-          lastName: responseUser.lastName,
-          firstName: responseUser.firstName,
-          job: responseUser.job,
-          role: responseUser.role,
-        };
-        dispatch(loginUser(userToRedux));
-        history.push('/lobby');
-      }
-    };
-
-    socket.emit('login', { user, room }, callback);
+
+    if (personalData.firstName.length > 0) {
+      console.log('send form');
+      const { firstName, lastName, jobPosition } = personalData;
+      console.log('room:', room);
+      console.log('url', url);
+
+      let userRole = ROLES.USER;
+      if (observer) userRole = ROLES.OBSERVER;
+      else if (isAdmin) userRole = ROLES.ADMIN;
+
+      const user: Member = {
+        userId: socket.id,
+        firstName,
+        lastName,
+        job: jobPosition,
+        role: userRole,
+      };
+
+      const callback = (response: ResponseFromSocket): void => {
+        console.log(response);
+
+        const { eventName, code, error: responseError, data } = response;
+
+        if (responseError) console.log(`${eventName}: ${code}: ${responseError}`);
+        else {
+          const { user: responseUser } = data;
+          dispatch(setCurrentRoom(responseUser.room));
+          const userToRedux = {
+            userId: responseUser.userId,
+            lastName: responseUser.lastName,
+            firstName: responseUser.firstName,
+            job: responseUser.job,
+            role: responseUser.role,
+          };
+          dispatch(loginUser(userToRedux));
+          history.push('/lobby');
+        }
+      };
+
+      socket.emit('login', { user, room }, callback);
+    } else {
+      setError('Please fill the form');
+      setIsError(true);
+    }
+  };
+
+  const handleNameChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    const v = e.target.value;
+    const len = v.length;
+    if (len < 3) {
+      setError('*the field "Name" must be at least 3 characters');
+      setIsError(true);
+    } else if (len > 12) {
+      setError('*the field "Name" must be no more than 12 characters');
+      setIsError(true);
+    } else {
+      setError('');
+      setIsError(false);
```

Comment:
maybe it's better to unite this with upper string?

Reply:
can you show what you mean?